app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -103,17 +102,20 @@ def audit_failure(err)
       end
 
       def identity_role
-        @role_class.by_login(
+        @identity_role ||= @role_class.by_login(
           jwt_identity,
           account: account
         )
       end
 
       # If there is no jwt identity so role and username are nil
       def audit_role_id
-        role = identity_role if @jwt_identity_initialized
-        username = jwt_identity if @jwt_identity_initialized
-        @role_id_class.new(
+        return @audit_role_id if @audit_role_id
+        if @jwt_identity
```

Comment:
```suggestion
        if jwt_identity
```

Reply:
@orenbm  this should be @jwt_identity so when we have failiure in validate_and_decode it won't start the jwt identity providing flow

Reply:
is it ok to have a failure in `validate_and_decode` should we continue to work in such a case? seems a bit odd programatically to rely on a failure

Reply:
if we have failiure in validate and decode we fail authentication. But we need to audit the failiure. The audit needs identity that we cant fetch because there is no token yet. So we want to audit nil.

This logic happens in audit only not in logic

Reply:
i see. please add a comment on this so future devs do not change it

Reply:
Thanks added